lib/rules/indent.js

```diff
@@ -608,9 +608,13 @@ module.exports = {
                 return elem !== null;
             });
 
-            // Skip if first element is in same line with this node
-            if (elements.length > 0 && elements[0].loc.start.line === node.loc.start.line) {
-                return;
+            // Skip if element is in same line with this node
+            if (elements.length > 0 && node.type === "ArrayExpression") {
+                elements.forEach(el => {
+                    if (el.loc.start.line === node.loc.start.line) {
+                        return;
+                    }
+                });
```

Comment:
@mysticatea this `forEach` is iterating through the `elements` and checking if the current element is on the same line as the start location of the node.
```
var x = [1,
    2,
    3
]
```
With the forEach example you are referring to, it would `return` on the `1` because the element is on the same line as the start node.